lib/async/http/faraday/adapter.rb

```diff
@@ -168,22 +173,29 @@ def call(env)
 				
 				def perform_request(env)
 					with_client(env) do |endpoint, client|
+						if headers = env.request_headers
+							headers = ::Protocol::HTTP::Headers[headers]
+							
+							# Use content-length to inform body length if given, but remove the header since it will be
+							# set for us later anyway, and not doing so could result in a duplicate content-length headers
+							# if capitalization differs
+							content_length = headers.delete("content-length")&.to_i
```

Comment:
This will give a content length of zero if the header was not set, e.g. `nil.to_i == 0`. I think it's better to write it like this:

```suggestion
							if value = headers.delete("content-length")
							  content_length = value.to_i
							end
```

Reply:
It will not be set to zero because of `&.`, which prevents calling `.to_i` if the callee is `nil`:

```
irb(main):001> nil&.to_i
=> nil
```

But if you think that syntax is unclear I can change it.